### ISSUE TITLE
Adding a large file using iroh-ctl fails due to provider records limit

### ISSUE THREAD

**Issue by rklaehn** [n0-computer/iroh#273]
When adding a large file using iroh-ctl add, it fails after some time due to an error about provider records.

```
rklaehn@M1-Pro iroh % cargo run --release --bin iroh-ctl add /Users/rklaehn/.diffusionbee/downloads/sd-v1-4.ckpt
    Finished release [optimized] target(s) in 0.19s
     Running `target/release/iroh-ctl add /Users/rklaehn/.diffusionbee/downloads/sd-v1-4.ckpt`
Error: status: Internal, message: "the store cannot contain any more provider records", details: [], metadata: MetadataMap { headers: {"content-type": "application/grpc", "date": "Mon, 03 Oct 2022 10:44:19 GMT", "content-length": "0"} }
```

The reason is that the provider table is currently in memory and size limited. Removing providing for each block fixes the issue.

This would probably be fixed by https://github.com/n0-computer/iroh/issues/251

**Comment by rklaehn** [comment]
@dignifiedquire I discussed a bit with @ramfox about how to solve this. go-ipfs "solves" this by having the option to only create a provider record for the root block of a large file. @ramfox said that you have some opinions about this. I would love to know what those are...

**Comment by dignifiedquire** [comment]
Step 1 is adding https://github.com/n0-computer/iroh/issues/251 so we don't have this limitation anymore. While it is nice to have the ability to only provide the root, it means you are forced to use a bitswap like exchange protocol, which is unfortunate.

**Comment by rklaehn** [comment]
We have a preliminary solution by just increasing the size of the mem store. It should be good enough for at least testing e.g. upload progress with large files. Fixed by #320
